qiskit/transpiler/__init__.py

```diff
@@ -40,714 +41,1132 @@
-
-   .. code-block:: python
-
-       from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
-       from qiskit.providers.fake_provider import FakeLagosV2
-
-       backend = FakeLagosV2()
-       pass_manager = generate_preset_pass_manager(3, backend)
-
-   which will generate a :class:`~.StagedPassManager` object for optimization level 3
-   targeting the :class:`~.FakeLagosV2` backend (equivalent to what is used internally
-   by :func:`~.transpile` with ``backend=FakeLagosV2()`` and ``optimization_level=3``).
-   You can use this just like working with any other :class:`~.PassManager`. However,
-   because it is a :class:`~.StagedPassManager` it also makes it easy to compose and/or
-   replace stages of the pipeline. For example, if you wanted to run a custom scheduling
-   stage using dynamical decoupling (via the :class:`~.PadDynamicalDecoupling` pass) and
-   also add initial logical optimization prior to routing you would do something like
-   (building off the previous example):
-
-   .. code-block:: python
-
-       from qiskit.circuit.library import XGate, HGate, RXGate, PhaseGate, TGate, TdgGate
-       from qiskit.transpiler import PassManager
-       from qiskit.transpiler.passes import ALAPScheduleAnalysis, PadDynamicalDecoupling
-       from qiskit.transpiler.passes import CXCancellation, InverseCancellation
-
-       backend_durations = backend.target.durations()
-       dd_sequence = [XGate(), XGate()]
-       scheduling_pm = PassManager([
-           ALAPScheduleAnalysis(backend_durations),
-           PadDynamicalDecoupling(backend_durations, dd_sequence),
-       ])
-       inverse_gate_list = [
-           HGate(),
-           (RXGate(np.pi / 4), RXGate(-np.pi / 4)),
-           (PhaseGate(np.pi / 4), PhaseGate(-np.pi / 4)),
-           (TGate(), TdgGate()),
-
-       ])
-       logical_opt = PassManager([
-           CXCancellation(),
-           InverseCancellation([HGate(), (RXGate(np.pi / 4), RXGate(-np.pi / 4))
-       ])
-
-
-       # Add pre-layout stage to run extra logical optimization
-       pass_manager.pre_layout = logical_opt
-       # Set scheduling stage to custom pass manager
-       pass_manager.scheduling = scheduling_pm
-
 
-   Then when :meth:`~.StagedPassManager.run` is called on ``pass_manager`` the
-   ``logical_opt`` :class:`~.PassManager` will be called prior to the ``layout`` stage
-   and for the ``scheduling`` stage our custom :class:`~.PassManager`
-   ``scheduling_pm`` will be used.
+      new_circ_lv3 = transpile(ghz, backend=backend, optimization_level=3)
+      plot_circuit_layout(new_circ_lv3, backend)
+
+
+It is completely possible to specify your own initial layout.  To do so we can
+pass a list of integers to :func:`qiskit.compiler.transpile` via the `initial_layout`
+keyword argument, where the index labels the virtual qubit in the circuit and the
+corresponding value is the label for the physical qubit to map onto:
```

Comment:
```suggestion
keyword argument where the index labels the virtual qubit in the circuit and the
corresponding value is the label for the physical qubit to map onto:
```
This has to be the hardest-to-describe `kwarg` input format in all Qiskit. Maybe something like "where each integer specifies a physical qubit index onto which the corresponding virtual qubit (following the order in `circuit.qubits` will be mapped"?

Reply:
Heh, I know the array of physical bits is Paul's preferred input format (I'm also fond of it too). But `initial_layout` supports like every way imaginable to describe the mapping. It might make more sense to use the dict format:

```
{ghz.qubits[0]: 3, ghz.qubits[1]: 4. ghz.qubits[2]: 2}
```

which is a bit easier to describe.